api/types/okta.go

```diff
@@ -0,0 +1,289 @@
+		return nil, trace.Wrap(err)
+	}
+	return o, nil
+}
+
+// GetUser returns the user that the actions will be applied to.
+func (o *OktaAssignmentV1) GetUser() string {
+	return o.Spec.User
+}
+
+func (o *OktaAssignmentV1) GetActions() []OktaAssignmentAction {
+	actions := make([]OktaAssignmentAction, len(o.Spec.Actions))
+
+	for i, action := range o.Spec.Actions {
+		actions[i] = action
+	}
+
+	return actions
+}
+
+// String returns the Okta assignment rule string representation.
+func (o *OktaAssignmentV1) String() string {
+	return fmt.Sprintf("OktaAssignmentV1(Name=%v, Labels=%v)",
+		o.GetName(), o.GetAllLabels())
+}
+
+// MatchSearch goes through select field values and tries to
+// match against the list of search values.
+func (o *OktaAssignmentV1) MatchSearch(values []string) bool {
+	fieldVals := append(utils.MapToStrings(o.GetAllLabels()), o.GetName())
+	return MatchSearch(fieldVals, values, nil)
+}
+
+// setStaticFields sets static resource header and metadata fields.
+func (o *OktaAssignmentV1) setStaticFields() {
+	o.Kind = KindOktaAssignment
+	o.Version = V1
+}
+
+// CheckAndSetDefaults checks and sets default values
+func (o *OktaAssignmentV1) CheckAndSetDefaults() error {
+	o.setStaticFields()
+	if err := o.Metadata.CheckAndSetDefaults(); err != nil {
+		return trace.Wrap(err)
+	}
+
+	if o.Spec.User == "" {
+		return trace.BadParameter("user must not be empty")
+	}
+
+	if len(o.Spec.Actions) == 0 {
+		return trace.BadParameter("actions is empty")
+	}
+
+	return nil
+}
+
+// OktaAssignmentAction is an individual action to apply to an Okta assignment.
+type OktaAssignmentAction interface {
+	// GetStatus returns the current status of the action.
+	GetStatus() string
+	// GetTargetType returns the target type of the action.
+	GetTargetType() string
+	// GetID returns the ID of the action target.
+	GetID() string
+}
+
+// GetStatus returns the current status of the action.
+func (o *OktaAssignmentActionV1) GetStatus() string {
+	switch o.Status {
+	case OktaAssignmentActionV1_PENDING:
+		return constants.OktaAssignmentActionStatusPending
+	case OktaAssignmentActionV1_SUCCESSFUL:
+		return constants.OktaAssignmentActionStatusSuccessful
+	case OktaAssignmentActionV1_FAILED:
+		return constants.OktaAssignmentActionStatusFailed
+	case OktaAssignmentActionV1_CLEANED_UP:
+		return constants.OktaAssignmentActionStatusCleanedUp
+	case OktaAssignmentActionV1_CLEANUP_FAILED:
+		return constants.OktaAssignmentActionStatusCleanupFailed
+	default:
+		return constants.OktaAssignmentActionStatusPending
+	}
+}
+
+// GetTargetType returns the target type of the action.
+func (o *OktaAssignmentActionV1) GetTargetType() string {
+	switch o.Target.Type {
+	case OktaAssignmentActionTargetV1_APPLICATION:
+		return constants.OktaAssignmentActionTargetApplication
+	case OktaAssignmentActionTargetV1_GROUP:
+		return constants.OktaAssignmentActionTargetGroup
+	default:
+		return constants.OktaAssignmentActionTargetApplication
```

Comment:
same question here, i guess whole point of adding uknown into proto enum is to handle it later